### PR TITLE
Patch for Select with aliases

### LINKED ISSUE #6: SQL select with Alaises
In original documentation , Select query with aliases is given the following explanation:
[https://github.com/allejo/PhpSoda/wiki/Working-with-SoQL-Queries](url)

I came across two issues while working with aliases:

1. When a select is given with alias, the $select statement generated has **white spaces** instead of **%20**. As a result, the HTTP request fails with response of 400 error code

eg: $selectWithAliases = array(
    "origin" => "origin_type"
);

URL formed: **https://.../.../...?$select=origin AS origin**
Expected URL: **https://.../.../...?$select=origin%20AS%20origin**

2. While trying the case "Not all column names must have an alias", the column given without alias are generated like **origin AS , file_type AS, year as publication_year** instead of origin, file_type, year as publication_year

eg: $selectWithAliases = array(
    "origin",
    "file_type",
    "year" => "publication_year"
);

URL formed: https://.../.../...?$select=origin AS ,file_type AS ,year AS publication_year
Expected URL: https://.../.../...?$select=origin,filer_name,year AS publication_year

A patch is required in line 169 of src/SoqlQuery.php to fix these 2 issues:

`$formattedValues[] = (is_string($key) && (!is_null($value))) ? rawurlencode(sprintf($format, trim($key), trim($value))) : (is_string($key) ? $key : $value);`

**(is_string($key) && (!is_null($value))) ? rawurlencode(sprintf($format, trim($key), trim($value))) : (is_string($key) ? $key : $value)** will handle the following cases:
1. Select with key and value ( column with alias )
2. Select with key and null value( column without alias while other column may come with alias)
3. Select with value ( column without alias in any column of select statement)

### PR DESCRIPTION
**Pull request by shravass**
Fixes #6 
Fixes for Working with SQL select with aliases

### CODE REVIEW

**Comment by allejo** [comment]
1. Would you mind terribly rebasing against master? I pushed an update to composer.json which should make some unit tests play nicer (hopefully)
2. Could you also add some unit tests to `tests/SoqlQueryTest.php` to ensure things stay working in the future?

**Comment by shravass** [comment]
Now 2 out of 3 have passed in travis-ci. Still one is failing
screen shot 2017-07-03 at 7 19 39 pm

**Comment by allejo** [comment]
Yea the hhvm one can be ignored, that's on me to fix

File: tests/SoqlQueryTest.php [hunk 1]
```diff
@@ -213,4 +213,45 @@ public function testMultipleGroupingQuery ()
 
         $this->assertContains($expected, (string)$soql);
     }
+
+    public function testSelectColumnsQueryWithPartialAliases ()
+    {
+        $soql = new SoqlQuery();
+        $soql->select("date_posted", "state", "sample_type" => "sample_value");
```
Comment (allejo) [inline tests/SoqlQueryTest.php]: Could you wrap these as an `array()`, not valid PHP otherwise :)

File: tests/SoqlQueryTest.php [hunk 2]
```diff
@@ -213,4 +213,45 @@ public function testMultipleGroupingQuery ()
...
+
+        $results = $this->dataset->getDataset($soql);
+
+        $this->assertArrayHasKey("date_posted", $results[0]);
+        $this->assertArrayHasKey("state", $results[0]);
+        $this->assertArrayHasKey("sample_value", $results[0]);
+        $this->assertArrayNotHasKey("sample_type", $results[0]);
+    }
+
+     public function testSelectColumnsQueryWithAliases ()
```
Comment (allejo) [inline tests/SoqlQueryTest.php]: Tiny little extra space before the public

**Review by allejo — CHANGES_REQUESTED** [review]

**Review by allejo — APPROVED** [review]

**Comment by shravass** [comment]
Sorry for the typo errors, they have been rectified. 
Thank you for your patience

**Comment by allejo** [comment]
Oh no worries at all! Thanks for finding the bug and fixing it 👍  I'll tag a new release in a bit

**Pull request merged**